onnxruntime/core/providers/cpu/rnn/rnn_helpers.h

```diff
@@ -249,27 +240,8 @@ void ExecuteLambdaInParallel(const std::string& name, TLambda lambda, int max, i
   }
 
   int totalTasks = (int)max / (step > 0 ? step : 1) + (max % step > 0 ? 1 : 0);
-  while (done != totalTasks) {
-  }
-#else
-  std::vector<std::future<void> > task_results{};
-  task_results.reserve(static_cast<size_t>(std::ceil(max / step)));
-
-  for (int i = 0; i < max; i += step) {
-    std::packaged_task<void()> task{std::bind(lambda, i)};
-    task_results.emplace_back(task.get_future());
-    ttp.RunTask(std::move(task));
-  }
-  try {
-    // wait for all and propagate any exceptions
-    for (auto& future : task_results)
-      future.get();
-  } catch (const std::exception& ex) {
-    LOGS(logger, ERROR) << name << " - exception running tasks: " << ex.what();
-    throw;
-  }
-#endif  // else part of #ifdef USE_EIGEN_THREADPOOLs
-#endif  // else part of #ifdef NOTHREADS
+  while (done != totalTasks);
+#endif
```

Comment:
We are planning to handle the execution exceptions in some generic way?

Reply:
Yes, but I did not modify this part of the code in this PR.